applications/FluidDynamicsBiomedicalApplication/custom_utilities/parabolic_profile_utilities.cpp

```diff
@@ -0,0 +1,241 @@
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Eduardo Soudah
+//                   Ruben Zorilla
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "processes/parallel_distance_calculation_process.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+
+// Application includes
+#include "fluid_dynamics_biomedical_application_variables.h"
+#include "parabolic_profile_utilities.h"
+
+namespace Kratos {
+
+void ParabolicProfileUtilities::CalculateWallParallelDistance(
+    ModelPart& rWallModelPart,
+    ModelPart& rFluidModelPart,
+    const std::size_t WallDistanceLevels)
+{
+    // Initialize the distance field to the geometry bounding box characteristic length
+    // Also initialize the BOUNDARY flag that will be used to mark the wall
+    const double char_length = CalculateBoundingBoxCharacteristicLength(rFluidModelPart);
+    block_for_each(rFluidModelPart.Nodes(), [char_length](NodeType& rNode){
+        rNode.Set(BOUNDARY, false);
+        rNode.SetValue(WALL_DISTANCE, char_length);
+    });
+
+    // Set a small negative distance value in all wall nodes
+    const double wall_d = -char_length * 1.0e-6;
+    KRATOS_ERROR_IF(std::abs(wall_d) < std::numeric_limits<double>::epsilon())
+        << "Auxiliary negative wall distance is close to zero. Check the domain dimensions." << std::endl;
+    block_for_each(rWallModelPart.Nodes(), [wall_d](NodeType& rNode){
+        rNode.Set(BOUNDARY, true);
+        rNode.GetValue(WALL_DISTANCE) = wall_d;
+    });
+    rWallModelPart.GetCommunicator().SynchronizeOrNodalFlags(BOUNDARY);
+    rWallModelPart.GetCommunicator().SynchronizeNonHistoricalDataToMin(WALL_DISTANCE);
+
+    // Calculate the distance in the first layer of nodes
+    // Note that these will be the values to be preserved by the parallel distance calculation below
+    auto aux_tls = std::make_pair(array_1d<double,3>(), array_1d<double,3>());
+    KRATOS_ERROR_IF(rWallModelPart.GetCommunicator().GlobalNumberOfConditions() == 0) << "Wall model part has no conditions." << std::endl;
+    block_for_each(rWallModelPart.Conditions(), aux_tls, [](
+        Condition& rCondition,
+        std::pair<array_1d<double,3>, array_1d<double,3>>& rTLS)
+    {
+        // Get TLS
+        auto& r_normal = std::get<0>(rTLS);
+        auto& r_wall_vect = std::get<1>(rTLS);
+
+        // Get condition geometry and calculate normal
+        auto& r_geom = rCondition.GetGeometry();
+        r_normal = r_geom.UnitNormal(0,GeometryData::IntegrationMethod::GI_GAUSS_1);
+
+        // Get parent geometry to search for the first layer of nodes
+        const auto& r_neighbours = rCondition.GetValue(NEIGHBOUR_ELEMENTS);
+        KRATOS_ERROR_IF(r_neighbours.size() == 0) << "No parent found in condition with id. " << rCondition.Id() << "." << std::endl;
```

Comment:
```suggestion
        KRATOS_ERROR_IF(r_neighbours.size() == 0) << "No parent found in condition with id " << rCondition.Id() << " in " <<  rWallModelPart.FullName() << "." << std::endl;
```

I would put the model part name also in the error.

Reply:
Cool.